# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLContext.h
# Commit: 073302d26c5e
# Full Hash: 073302d26c5e1ac8229fca1b32656de68756e0f2
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 09:34:54
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - Create a separate WebGLContext::PresentInto function for WebXR. r=jgilbert
#   
#   - Do not apply mIsCanvasDirty bailout when presenting from a opaque FBO
#   back buffer.
#   - Create a separate WebGLContext::PresentInto with more specific XR
# ==============================================================================

diff -r 7c94d37c446e -r 073302d26c5e dom/canvas/WebGLContext.h
--- a/dom/canvas/WebGLContext.h	Thu Jun 11 06:37:36 2020 +0000
+++ b/dom/canvas/WebGLContext.h	Thu Jun 11 06:37:38 2020 +0000
@@ -101,7 +101,7 @@
 namespace layers {
 class CompositableHost;
 class SurfaceDescriptor;
-}
+}  // namespace layers
 
 namespace webgl {
 class AvailabilityRunnable;
@@ -483,6 +483,7 @@
   // Present to compositor
  private:
   bool PresentInto(gl::SwapChain& swapChain);
+  bool PresentIntoXR(gl::SwapChain& swapChain, const gl::MozFramebuffer& fb);
 
  public:
   void Present(WebGLFramebuffer*, layers::TextureType);
